tests/test_bucket_versioning.py

```diff
@@ -0,0 +1,396 @@
+        version_list = self.list_all_versions_of_the_object(
+            c_scope_s3client, bucket, self.obj_name
+        )
+        for v in version_list:
+            log.info(v)
+
+        # delete first version of the object
+        del_resp = c_scope_s3client.delete_object(
+            bucket, self.obj_name, VersionId=version_list[0]
+        )
+        assert (
+            del_resp["Code"] == 204
+        ), f"delete_object resulted in an unexpected response: {del_resp}"
+
+        # List all versions of the object after delete operaion
+        version_list = self.list_all_versions_of_the_object(
+            c_scope_s3client, bucket, self.obj_name
+        )
+        for v in version_list:
+            log.info(v)
+        assert (
+            len(version_list) == 1
+        ), f"Deletion of versioned object with version id failed"
+        log.info("Successfully deleted versioned object with version id")
+
+    @tier1
+    def test_delete_versioned_object(self, c_scope_s3client):
+        """
+        Test list version ids of the object from versioned bucket:
+        1. Create regular bucket
+        2. Enable versioning on bucket
+        3. Upload object in versioned bucket
+        4. Delete object without specifying version of the object
+        5. Verify that the delete marker appears in version list
+        """
+        # Create regular bucket and enable versioning on it
+        bucket = self.setup_versioned_bucket(c_scope_s3client)
+
+        # Upload data with object_key in versioned bucket
+        log.info("Uploading data in versioned bucket")
+        response = c_scope_s3client.put_object(bucket, self.obj_name, self.obj_data)
+        assert (
+            response["Code"] == 200
+        ), f"put_object failed with response code {response['Code']}"
+
+        # delete the object
+        del_response = c_scope_s3client.delete_object(bucket, self.obj_name)
+        assert (
+            del_response["Code"] == 204
+        ), f"delete_object resulted in an unexpected response: {del_response}"
+
+        # Verify delete marker is added in version list of the object after delete operaion
+        response = c_scope_s3client.list_object_versions(bucket)
+        assert response["DeleteMarkers"][0]["VersionId"] == del_response.get(
+            "VersionId"
+        ), f"Delete marker is not added for object {self.obj_name}"
+        log.info("Delete marker added successfully to the deleted object")
+
+    @tier1
+    def test_restore_deleted_versioned_object(self, c_scope_s3client):
+        """
+        Test list version ids of the object from versioned bucket:
+        1. Create regular bucket
+        2. Enable versioning on bucket
+        3. Upload object in versioned bucket
+        4. Delete object without specifying version of the object
+        5. Verify that the delete marker appears in version list
+        6. Restore object by deleting delete marker for the object
+        """
```

Comment:
Merge this test with `test_delete_versioned_object` - the only difference is the final step where we restore the object by deleting the delete marker

Reply:
Fixed